### DISCUSSION TITLE
Idea: Add support for shell offset to shell property card so not all shells have to be midplane

### DISCUSSION THREAD

**Discussion by kv-memerson** [OpenRadioss/OpenRadioss#999] [Ideas]
Coming from Abaqus and Dyna, one thing that I really miss is being able to use nloc (dyna) or SPOS/SNEG (abaqus) to specify if the input mesh is the midplane, positive, or negative side of the shell element. Meshing everything with midplane can be a pain and this feature could alleviate that.

All in all, loving openradioss so far!

**Comment by MarianBulla** [comment]
Hi, 
Thank you very much for your kind words :-)
Yes, it is also possible with OpenRadioss, to shift the midplane of shells. 

Please, have a look at /PROP/TYPE51.

Thanks and best regards,
Marian

**Reply by kv-memerson** [reply]
Thank you for your quick reply Marian! I will give that a try. I did not think to define it with a ply. Probably it would just be one ply and the angle would be irrelevant if it is an isotropic material model right?

Thanks again for your advice!

**Reply by MarianBulla** [reply]
Yes, exactly. It works also with just 1 ply and an isotropic material. 
I'm using it for modelling of windshield glass (shells with offset) attached to the PVB foli (solids).
Both are isotropic materials.

Anyway, pay attention on that and use it with case, because such an offset can lead to unbalanced stiffness in case of some loadcases with unexpected results.

Thanks and best regards,
Marian

**Reply by kv-memerson** [reply]
Hey Marian,

I took a look at this and tried to modify our model. One thing we have run into is that we have a non-uniform shell thickness distribution defined on a per element basis. It seems that with the ply model, the offset must be set manually but with a varying thickness profile, the ply offset would need to be different for each element. Do you have any ideas on how we could make the nonuniform thickness profile work with the ply method?

Thanks!!
Mike Emerson

**Reply by MarianBulla** [reply]
Hey Mike,
will the I_pos option 3 or 4 (in the /PROP/TYPE51 property) not help ?

Thanks and best regards,
Marian

**Reply by kv-memerson** [reply]
Ahh I see, I will give that a try, thanks again for your quick help on all of this Marian!

best,
Mike Emerson

**Reply by MarianBulla** [reply]
You're always welcome :-)
If it will not help you, then we have to think about a script, which will do the job.

Thanks and best regards,
Marian

**Reply by kv-memerson** [reply]
Had some success with this with the I_pos 3/4 on the property! Thank you for all the help with that. We got the model running and things are beginning to take shape. As one follow up question, I think we may have found a bug (or maybe we are missing a control setting somewhere). When using the ply with thickness offset like I_pos = 3 for example, it seems like the contact gap is still being calculated as half of the element thickness from the node locations (not accounting for the shell offset). The contact we tried was INTER TYPE19 and we tried all values of Igap. Maybe a different contact methodology or control would be required for it to calculate the gap with consideration to the ply location relative to the shell. Any ideas for that?

**Reply by MarianBulla** [reply]
Hi,
yes, you are right. This (the shell offset) not considered in most of the contact interfaces. Has to be set manually, by adjusting the contact GAP value, within the contact interface.

Maybe only in the INTER Type21, which is used for forming/stamping, where one part is rigid.

It's a very good sugestion for further improovemnt.

Thanks and best regards,
Marian

**Reply by MahaHachani** [reply]
Hi, 
For different contacts in RADIOSS, the gap is computed by considering the midplane of shells and not considering the offset.
Further developments have to be done to take into account shell offset in contact.
Thank you 
Best Regards
Maha
